### PR TITLE
adnexo + Cleanup

### PR DESCRIPTION
**Pull request by J2thatsme**
added ax-dist, cleaned up ax-opto & changed some metadata

### CODE REVIEW

File: types/adnexo/ax-dist/invalid_measurement.schema.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "$id": "https://akenza.io/adnexo/ax-dist/invalid_measurement.schema.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Invalid measurement",
+  "type": "object",
+  "properties": {
+    "measVariance": {
```
Comment (mariusgiger) [inline types/adnexo/ax-dist/invalid_measurement.schema.json:7]: measurementVariance?

File: types/adnexo/ax-dist/measurement.schema.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  "title": "Measurement",
+  "type": "object",
+  "properties": {
+    "distance": {
+      "title": "Distance",
+      "unit": "cm",
+      "type": "number",
+      "description": "Distance measured between the sensor and the surface"
+    },
+    "percent": {
```
Comment (mariusgiger) [inline types/adnexo/ax-dist/measurement.schema.json:13]: fillLevel?

File: types/adnexo/ax-dist/measurement.schema.json [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      "title": "Percent",
+      "unit": "%",
+      "type": "number",
+      "description": "Percentage of the container fill"
```
Comment (mariusgiger) [inline types/adnexo/ax-dist/measurement.schema.json:17]: Fill level of the container?

File: types/adnexo/ax-dist/meta.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "ax-dist",
+  "version": "1.0.0",
+  "manufacturer": "Adnexo",
+  "url": "https://www.adnexo.ch/de/unser-angebot/hardware/#ax-dist",
+  "description": "The ax-dist is a distance measuring sensor with LoRaWAN wireless technology. It measures distances of up to 5m and can be used for a variety of applications such as parking, fill level or seat occupation measurement. The ax-dist has a battery life span of up to five years depending on the settings and can be used both indoors and outdoors.",
```
Comment (mariusgiger) [inline types/adnexo/ax-dist/meta.json:6]: lifespan is one word

File: types/adnexo/ax-dist/uplink.js [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+  } else if (port === 4) {
+    let pointer = 0;
+
+    while (pointer < payload.length) {
+      const header = Bits.bitsToUnsigned(bits.substr(pointer * 4, 8));
+      pointer += 2;
+
+      switch (header) {
+        case 0x01:
+          data.appEui = payload.substr(pointer, 16);
```
Comment (mariusgiger) [inline types/adnexo/ax-dist/uplink.js:43]: this is not in the schema, why is this even reported by the sensor?

File: types/adnexo/ax-dist/uplink.js [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+    if (distance !== 6553.5) {
+      data.distance = distance;
+    } else {
+      data.distance = null;
+    }
+
+    if (event.device !== undefined && data.distance !== undefined) {
+      if (event.device.customFields !== undefined) {
+        const scaleLength = Number(
+          event.device.customFields["Height of the tank in cm"],
```
Comment (mariusgiger) [inline types/adnexo/ax-dist/uplink.js:122]: better use custom fields that use camelCase: tankHeightCm in order to ease programmatic access

**Review by mariusgiger — COMMENTED** [review]
looks good, a few comments

File: types/adnexo/ax-dist/uplink.js (hunk 6 above)
Comment (J2thatsme) [reply types/adnexo/ax-dist/uplink.js:122]: Yeah but it's harder to understand for the people which need to fill in those values afterwards. 
image
For me either way is fine.

Comment (mariusgiger) [reply types/adnexo/ax-dist/uplink.js:110]: we should maybe add a description to custom fields that will allow writing sentence like you did? could you propose that in Canny?

**Review by mariusgiger — COMMENTED** [review]
looks good, I would still rename the custom fields to be easily used in code

File: types/adnexo/ax-dist/uplink.js [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
```
Comment (mariusgiger) [inline types/adnexo/ax-dist/uplink.js:110]: first make sure that the field with the given name exists before parsing it as number

**Review by mariusgiger — APPROVED** [review]

**Pull request merged**